KOALA-2817: FHIR - MedicationDispense (#799)
canvas-medical/documentation: collections/_api/medicationdispense.md

            description: Indicates the medication order that is being dispensed against.
            type: array[json]
            attributes:
              - name: reference
                type: string
                description: The reference string of the MedicationRequest in the format of `"MedicationRequest/3423a69c-618d-4cbe-861a-54c60f48744e"`.
              - name: type
                type: string
                description: Type the reference refers to (e.g. "MedicationRequest").
          - name: type
            description: Indicates the type of dispensing event that is being performed.
            type: json
            attributes:
              - name: coding
                description: Code defined by a terminology system.
                type: array[json]
                attributes: 
                  - name: system
                    description: The system url of the coding.
                    type: string
                  - name: code
                    description: The code of the dispense type.
                    type: string
                  - name: display
                    description: The display name of the coding.

>                     type: string

Reviewer: The `type` attribute needs to be changed. We aren't supporting the coding here and will only fill in the `text` attribute.

          - name: quantity
            description: The amount of medication that has been dispensed. Includes unit of measure.
            type: json
            attributes:
              - name: value
                type: decimal
                description: Numerical value of the quantity.
              - name: unit
                type: string
                description: Unit representation (e.g., "Tablet", "ml", "mg").
              - name: system
                type: string
                description: System that defines the coded form of the unit.
              - name: code
                type: string
                description: Coded form of the unit.
          - name: whenHandedOver

>             description: When the medication was handed over to the patient or their representative. SHALL be present if the status is "completed".

Reviewer: Let's simplify the description here. Also, we typically won't include SHALL phrases because that's a level of detail we don't want to get into in our documentation.

```suggestion
            description: When the medication was handed over to the patient.
```

            type: datetime
          - name: dosageInstruction
            description: Indicates how the medication is to be used by the patient.
            type: array[json]
            attributes:
                - name: text
                  type: string
                  description: Free text dosage instructions.
                - name: timing
                  type: json
                  description: When medication should be administered.
                  attributes:
                    - name: event
                      type: array[string]

>                       description: Identifies specific times when the event takes place.

Reviewer: Let's be more specific here.

```suggestion
                      description: Identifies the specific times when the medication should be administered.
```

                - name: doseAndRate
                  type: array[json]
                  description: Amount of medication administered.
                  attributes:
                    - name: doseQuantity
                      type: json
                      description: Amount of medication per dose.

>                       attributes:

Reviewer: We need to add the `value` attribute.

                        - name: unit
                          description: Unit representation. 
                          type: string
        search_parameters:
          - name: _id
            description: The identifier of the MedicationDispense.
            type: string
          - name: patient

>             description: The patient reference associated to the Medication Dispense in the format `Patient/a39cafb9d1b445be95a2e2548e12a787`.

Reviewer: Grammar correction here (replacing to with "with"). If you have time and can make the correction to MediationRequest too, please do.

```suggestion
            description: The patient reference associated with the MedicationDispense in the format `Patient/a39cafb9d1b445be95a2e2548e12a787`.
```

            type: string
        endpoints: [read, search]
        read:
          description: Read a MedicationDispense resource.
          responses: [200, 401, 403, 404]
          example_request: medicationdispense-read-request
          example_response: medicationdispense-read-response
        search:
          description: Search for MedicationDispense resources.
          responses: [200, 400, 401, 403]
          example_request: medicationdispense-search-request
          example_response: medicationdispense-search-response
---

<div id="medicationdispense-read-request">
{%  include read-request.html resource_type="MedicationDispense" %}
</div>

<div id="medicationdispense-read-response">

  {% tabs medicationdispense-read-response %}

    {% tab medicationdispense-read-response 200 %}
```json

> {

Reviewer: Please apply any changes to this example to the other examples in this file as well.

    "status": "completed",
    "medicationCodeableConcept": {
        "coding": [
            {
                "system": "http://www.fdbhealth.com/",
                "code": "244899",
                "display": "lisinopril 10 mg tablet"
            },
            {
                "system": "http://www.nlm.nih.gov/research/umls/rxnorm",
                "code": "314076",
                "display": "lisinopril 10 mg tablet"
            }
        ]
    },
    "subject": {
        "reference": "Patient/6cb2a409334943c2b48f1686dc739f11",
        "type": "Patient"
    },
    "performer": [
        {
            "actor": {
                "reference": "Practitioner/6c20b7152cf7421791c5ab4113060b3f",
                "type": "Practitioner"
            }
        }
    ],
    "authorizingPrescription": [
        {
            "reference": "MedicationRequest/3423a69c-618d-4cbe-861a-54c60f48744e",
            "type": "MedicationRequest"
        }
    ],
    "type": {
        "coding": [
            {
                "system": "http://terminology.hl7.org/CodeSystem/v3-ActCode",
                "code": "RFP",
                "display": "Refill - Part Fill"
            }
        ]

>     },

Reviewer: This needs to be changed — remove the coding and add the `text` attribute that is hard-coded to `Office-supplied`.

    "quantity": {
        "value": 30,
        "unit": "Tablet",
        "system": "http://unitsofmeasure.org",
        "code": "mL"

>     },

Reviewer: We're only providing `value` here, so the other attributes need to be removed.

    "whenHandedOver": "2023-09-21T18:35:00.000+00:00",
    "dosageInstruction": [
        {
            "text": "take 1 daily",
            "timing": {
              "event": [

>                 "2025-09-10T10:44:13.842370+00:00"

Reviewer: Let's have this datetime match `whenHandedOver`, since that is how the API will behave.

              ]
            },
            "doseAndRate": [
                {

>                     "doseQuantity": {

Reviewer: We need to add the `value` attribute.

I see that you modeled this after MedicationRequest, which also has the same mistake. If you have time, it would also be nice to fix those docs as well.
